x-pack/plugins/synthetics/public/legacy_uptime/components/monitor_management/monitor_list/monitor_detail_flyout.tsx

```diff
@@ -0,0 +1,286 @@
+  EuiLoadingSpinner,
+  EuiSelect,
+  EuiSpacer,
+  EuiSwitch,
+  EuiTitle,
+  useEuiTheme,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import moment from 'moment';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { FETCH_STATUS, useFetcher } from '@kbn/observability-plugin/public';
+import { DecryptedSyntheticsMonitorSavedObject } from '../../../../../common/types';
+import { fetchPings, getMonitor } from '../../../state/api';
+import { ClientPluginsStart } from '../../../../plugin';
+import { useStatusByLocation } from '../../../../hooks/use_status_by_location';
+// TODO: remove any
+const FlyoutHeadingTable = styled<any>(EuiBasicTable)`
+  .euiTableRowCell {
+    border-bottom: 1px solid transparent;
+    border-top: 1px solid transparent;
+  }
+  .euiTable {
+    width: inherit;
+  }
+`;
+
+const BoldItem = styled(EuiFlexItem)`
+  font-weight: bold;
+`;
+
+const STATUS_COLUMN_NAME = i18n.translate('xpack.synthetics.monitorList.statusColumnName', {
+  defaultMessage: 'Status',
+});
+
+const LOCATION_COLUMN_NAME = i18n.translate('xpack.synthetics.monitorList.locationColumnName', {
+  defaultMessage: 'Location',
+});
+
+const DURATION_HEADER_TEXT = i18n.translate('xpack.synthetics.monitorList.durationHeaderText', {
+  defaultMessage: 'Duration',
+});
+
+const MONITOR_DETAILS_HEADER_TEXT = i18n.translate(
+  'xpack.synthetics.monitorList.monitorDetailsHeaderText',
+  {
+    defaultMessage: 'Monitor Details',
+  }
+);
+
+const ENABLED_ITEM_TEXT = i18n.translate('xpack.synthetics.monitorList.enabledItemText', {
+  defaultMessage: 'Enabled',
+});
+
+const LOCATIONS_ITEM_TEXT = i18n.translate('xpack.synthetics.monitorList.locationsItemText', {
+  defaultMessage: 'Locations',
+});
+
+const NAME_ITEM_TEXT = i18n.translate('xpack.synthetics.monitorList.nameItemText', {
+  defaultMessage: 'Name',
+});
+
+const MONITOR_ID_ITEM_TEXT = i18n.translate('xpack.synthetics.monitorList.monitorIdItemText', {
+  defaultMessage: 'Monitor ID',
+});
+
+const CLOSE_FLYOUT_TEXT = i18n.translate('xpack.synthetics.monitorList.closeFlyoutText', {
+  defaultMessage: 'Cancel',
+});
+
+const GO_TO_MONITOR_LINK_TEXT = i18n.translate('xpack.synthetics.monitorList.goToMonitorLinkText', {
+  defaultMessage: 'Go to monitor',
+});
+export function MonitorDetailFlyout(props: { id: string; location: string; onClose: () => void }) {
+  const { id } = props;
+  const theme = useEuiTheme();
+  const [location, setLocation] = useState<string>(props.location);
+  const { observability } = useKibana<ClientPluginsStart>().services;
+  const { ExploratoryViewEmbeddable } = observability;
+  const {
+    data: monitorSavedObject,
+    error,
+    status,
+  } = useFetcher<Promise<DecryptedSyntheticsMonitorSavedObject | undefined>>(
+    () => getMonitor({ id }),
+    [id]
+  );
+  const { data: detailData, status: detailStatus } = useFetcher(() => {
+    return fetchPings({
```

Comment:
I'm not sure if we should be using legacy uptime fetchPings. I believe our goal is to use the data plugin or the useEsQuery hook moving forward. Thoughts @shahzad31 ?

Reply:
Also, this legacy query likely filters by `monitor.id`, whereas we'll want to filter by `config_id` moving forward.